Post community review

nautolycus opened the pull request:
Electron diffraction CIF dictionary revised extensively following community discussions. Ancillary documentation updated accordingly. New proto-dictionary draft_new_items.dic also created as scratchpad for possible extensions to the imgCIF dictionary.

Note validation of cif_ed.dic will fail on _units.code values reciprocal_angstroms_per_pixel and electrons_per_angstrom. These will be submitted as a PR to templ_enum.cif pending acceptance of this new dictionary.

nautolycus committed:
Terminology updated in recommendations.md

nautolycus committed:
New ancillary documentation to identify essential data collection metadata

nautolycus committed:
Added mapping to CIF data items to Zenodo template

nautolycus committed:
Updated zenodo_template.md to latest template version

nautolycus committed:
Added docs/frames_example.cif as example imgCIF-style file

nautolycus committed:
Added draft_new_items.dic as a set of possible new definitions.

nautolycus committed:
Amended description of probe formation

nautolycus committed:
Added new draft items to Zenodo template mappings

nautolycus committed:
Updated draft_new_items.dic and docs/recommendations.md

nautolycus committed:
Fixed silly typo in draft_new_items.dic

nautolycus committed:
Added CIF2.0 header to draft_new_items.dic

nautolycus committed:
Moved potential cif core extension terms from draft_new.items.dic to cif_ed.dic

nautolycus committed:
Updated README.md to explain purpose of new_draft_items.dic file.

nautolycus committed:
Added _diffrn_detector.distance_sample to cif_ed.dic

nautolycus committed:
Changed _diffrn_source.ed_probe_formation to _diffrn_source.ed_probe_type
and _diffrn.precession_angle to _diffrn.precession_semi_angle
(and corresponding *_su term).

nautolycus committed:
Updates to draft dictionary following community discussion; ed.pdf and recommendations.md updated accordingly.

nautolycus committed:
Further changes to cif_ed.dic, docs/ed.pdf and docs/recommendations.md,
mostlt relating to changes from _refine_diff.density_* to *.potential_*

vaitkus commented on draft_new_items.dic:
```diff
@@ -0,0 +1,131 @@
+    _definition.update            2024-07-15
+    _description.text
+;
+    The scale factor for this frame.
```
```suggestion
    Standard uncertainty of _diffrn_scan_frame.scale_factor.
```

vaitkus commented on cif_ed.dic:
```diff
@@ -313,261 +417,288 @@ save_diffrn_source.size
+;
+    _name.category_id             refine
+    _name.object_id               diffraction_theory
+    _type.purpose                 Encode
```
If I understand correctly, the 'Encode' purpose is used to note that the values are machine parsable, i.e. have some kind of additional internal structure, e.g. an InChI string. I suggest that here and in other definitions with the `_enumeration_set.state` attribute the 'State' type purpose be used.

nautolycus replied:
Thank you for that explanation. I must have done an initial cut-and-paste without much thought. I'll update the PR with a new commit of the dictionary.

vaitkus commented on docs/recommendations.md:
````diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+
+```
+_exptl_crystal.mosaicity              0.050
+_exptl_crystal.method                 'From rocking curve, width 0.00070'
````
I could not find the definition of this item. Maybe `_exptl_crystal.mosaic_method` is meant instead?

vaitkus commented on docs/recommendations.md:
````diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+(_b_)
+```
+_diffrn_detector.ed_calibration_constant      0.008259
+_diffrn_source.convergence_angle              0.000(5)
````
According to the newest version of the dictionary, values of the `_diffrn_source.convergence_angle` data item are not allowed to have standard uncertainty (S.U.).

vaitkus replied:
I saw that  the s.u. got removed in one of the later commits so the example no longer conflicts with the dictionary definition. So the mistake was in the example and not the dictionary, i.e. s.u. values will never be required for the `_diffrn_source.convergence_angle` data item?

nautolycus replied:
Sorry not to have commented on this last time. Yes: the conclusion was that the angle is set by the instrument manufacturer and is not directly measured (_type.source Assigned), so no s.u. is calculable.

vaitkus replied:
Thank you for clarifying.

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+   2  axis_beta     0.053      0.0
+   2  axis_omega    0.101      0.0
+   3  axis_u        1.00000    0.0
+   3  axis v       -0.43528    0.0
```
```suggestion
   3  axis_v       -0.43528    0.0
```

vaitkus commented on docs/recommendations.md:
````diff
@@ -2,6 +2,8 @@
+
 Most of the definitions in the core CIF dictionary can be applied to a description of a 3D ED structure determination experiment. An example of good practice in describing the relevant experimental metadata follows, with some guidance for individual data items. (The current example is adapted from the published structure of Andrusenko, I., Potticary, J., Hall, S. R. & Gemmi, M. (2020). _Acta Crys ...
 
 ```
````
I example below is slightly outdated. I cannot mark the specific lines so a bullet list will have to do:
- `_diffrn_precession.angle` should be changed to `_diffrn.precession_semi_angle`.
- `_diffrn_measurement.integration` should be changed to `_diffrn_measurement.rotation_mode`.
- `_diffrn_radiation.wavelength` should be changed to `_diffrn_radiation_wavelength.value` (`_diffrn_radiation.wavelength` alias is not currently defined in the CIF_CORE dictionary).

Also, section "`_diffrn_source.voltage` and `_diffrn_radiation.wavelength`" of this document claim that the `_diffrn_radiation.wavelength` is expected to include SU so it would be nice to also do that in the given example if possible.

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
 This free-text field, already present in the core dictionary, can be used to describe the appropriate measure of the beam size at the sample. For selected-aperture electron diffraction, this should be the beam width corresponding to the aperture setting, and not the total beam width.
 
-## `_diffrn_source.voltage` and `_diffrn_radiation.wavelength`
+### `_diffrn_source.voltage` and `_diffrn_radiation.wavelength`
 
 Both should be supplied for electron diffraction studies. The value of `_diffrn_radiation.wavelength` is expected to include a standard uncertainty to express the expected wavelength spread.
```
The `_diffrn_radiation.wavelength` data name is not currently defined in the core dictionary. Alternatives:
- `_diffrn_radiation_wavelength.value` (main name)
- `_diffrn_radiation_wavelength` (alias, old name)
- `_diffrn_radiation_wavelength.wavelength` (alias, probably from mmCIF/PDBx)

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+### `_diffrn_measurement.rotation_mode`
 
-The method of collecting data from regions of reciprocal space should take one of the values 'beam-tilt', 'rotation' (or its synonym 'continuous') or 'stepwise' according to the technique used. The beam-tilt method is sometimes called 'the rotation method' in 3D electron diffraction.
+The method of collecting data from regions of reciprocal space should take one of the values 'rotation' or 'stepwise' according to the technique used. Continuous rotation is the usual method in X-ray crystallography. 'Stepwise' mode icludes cases where the sample is panned or tilted in discrete steps and/or the electron beam precesses around a conical surface normal to the sample.
```
```suggestion
The method of collecting data from regions of reciprocal space should take one of the values 'rotation' or 'stepwise' according to the technique used. Continuous rotation is the usual method in X-ray crystallography. 'Stepwise' mode includes cases where the sample is panned or tilted in discrete steps and/or the electron beam precesses around a conical surface normal to the sample.
```

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
 The crystal colour is normally required by checkCIF, but can be difficult or impossible to determine for nanocrystals. In such a case the special CIF character `?` (unquoted) may be used with the meaning 'unknown', or a value of 'undetermined' may be given.
 
-## `_diffrn_measurement.tracking` and `_diffrn_measurement.tracking_method`
+### `_diffrn_measurement.tracking` and `_diffrn_measurement.tracking_method`
```
```suggestion
### `_diffrn_measurement.sample_tracking` and `_diffrn_measurement.sample_tracking_method`
```

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+### `_diffrn_measurement.tracking` and `_diffrn_measurement.tracking_method`
 
-`_diffrn_measurement.tracking` is a simple yes/no code to indicate if some tracking method was used to maximize the time that the sample is illuminated by the crystal, if the crystal drifts out of the beam during data collection. Details of the method can be given as free text in the `_diffrn_measurement.tracking_method` field.
+`_diffrn_measurement.tracking` is a simple yes/no code to indicate if some tracking method was used to maximize the time that the sample is illuminated by the crystal, if the crystal drifts out of the beam during data collection. Details of the method can be given as free text in the `_diffrn_measurement.tracking_method` field. _A literature reference may also be supplied with the data item `_com ...
```
```suggestion
`_diffrn_measurement.sample_tracking` is a simple yes/no code to indicate if some tracking method was used to maximize the time that the sample is illuminated by the crystal, if the crystal drifts out of the beam during data collection. Details of the method can be given as free text in the `_diffrn_measurement.sample_tracking_method` field. _A literature reference may also be supplied with the data item `_computing.sample_tracking`._
```

vaitkus commented on docs/recommendations.md:
````diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+
+```
+
+### `_exptl_crystal.mosaicity` and `_exptl_crystal.mosaicity_method`
````
```suggestion
### `_exptl_crystal.mosaicity` and `_exptl_crystal.mosaic_method`
```

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+
+### `_exptl_crystal.mosaicity` and `_exptl_crystal.mosaicity_method`
+
+The mosaicity (in degrees) and method of determination may be supplied. If known, the average mosaic block size in ångström units may also be given using `_exptl_crystal.mosaic_block_size`.
```
The `_exptl_crystal.mosaic_method` data item is listed in the section title, but not explicitly described.

vaitkus replied:
My comment seems to be a false-positive. The first sentence does mention the "method of determination".

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
+
+### `_diffrn_source.ed_diffracting_area_selection`
+
+A code specifying whether the diffraction region was selected by the size of the beam or whether selected-area electron diffraction technique used to illuminate the appropriate region of the sample.
```
This sentence is a bit difficult to parse.

vaitkus commented on docs/recommendations.md:
```diff
@@ -25,63 +27,157 @@ _diffrn_measurement.integration        stepwise
-## `_diffrn_measurement.method_precession`
+### `_diffrn_measurement.method_precession`
 
 If beam precession was used, this data item should be present and have the value 'yes' or 'y'. The precession angle should be given as the value of `_diffrn_precession.angle`, including standard uncertainty where possible.
```
```suggestion
If beam precession was used, this data item should be present and have the value 'yes' or 'y'. The precession angle should be given as the value of `_diffrn.precession_semi_angle`, including standard uncertainty where possible.
```

nautolycus committed:
Update draft_new_items.dic

Description of _diffrn_scan_frame.scale_factor_su fixed.

Co-authored-by: Antanas Vaitkus <antanas.vaitkus90@gmail.com>

nautolycus committed:
Items in cif_ed.dic with enumerate state given _type.purpose 'State' and
docs/recommendations.md updated with many helpful editorial improvements
suggested by Antanas Vaitkus.

vaitkus commented on docs/recommendations.md:
```diff
@@ -20,68 +22,162 @@ _diffrn_detector.make                 'ASI Timepix'
+### `_diffrn_measurement.sample_tracking` and `_diffrn_measurement.sample_tracking_method`
 
-`_diffrn_measurement.tracking` is a simple yes/no code to indicate if some tracking method was used to maximize the time that the sample is illuminated by the crystal, if the crystal drifts out of the beam during data collection. Details of the method can be given as free text in the `_diffrn_measurement.tracking_method` field.
+`_diffrn_measurement.sample_tracking` is a simple yes/no code to indicate if some tracking method was used to maximize the time that the sample is illuminated by the crystal, if the crystal drifts out of the beam during data collection. Details of the method can be given as free text in the `_diffrn_measurement.tracking_method` field. _A literature reference may also be supplied with the data ite ...
```
```suggestion
`_diffrn_measurement.sample_tracking` is a simple yes/no code to indicate if some tracking method was used to maximize the time that the sample is illuminated by the crystal, if the crystal drifts out of the beam during data collection. Details of the method can be given as free text in the `_diffrn_measurement.sample_tracking_method` field. _A literature reference may also be supplied with the data item `_computing.sample_tracking`._
```

nautolycus committed:
Fix residual typo in docs/recommendations.md

Co-authored-by: Antanas Vaitkus <antanas.vaitkus90@gmail.com>
